File: solutions/539.minimum-time-difference/minimum-time-difference.java
class Solution {
    public int findMinDifference(List<String> timePoints) {
        PriorityQueue<Integer> pq = new PriorityQueue<Integer>();
        for(String timePoint : timePoints) 
            pq.offer(format(timePoint));
        int first = pq.peek(), pre = pq.poll(), min = Integer.MAX_VALUE;
        while(!pq.isEmpty()) {
            min = Math.min(min, pq.peek() - pre); 
            pre = pq.poll();
        }
        return Math.min(min, 1440 - pre + first);
    }
    private int format(String time) {
        int digit1 = time.charAt(0);
        int digit2 = time.charAt(1);
        int digit3 = time.charAt(3);
        int digit4 = time.charAt(4);
        return (digit1 * 10 + digit2) * 60 + digit3 * 10 + digit4;
    }
}

File: solutions/539.minimum-time-difference/minimum-time-difference.cpp
class Solution {
public:
    int findMinDifference(vector<string>& timePoints) {
        sort(timePoints.begin(), timePoints.end()); 
        int n = timePoints.size();
        int minDiff = INT_MAX;
        for(int i = 1; i < n; i++) 
            minDiff = min(minDiff, getDiff(timePoints[i - 1], timePoints[i]));
        return min(minDiff, 1440 - getDiff(timePoints[0], timePoints[n - 1]));
    }
private:
    int getDiff(string pre, string next) {
        int h1 = stoi(pre.substr(0, 2));
        int m1 = stoi(pre.substr(3, 2));
        int h2 = stoi(next.substr(0, 2));
        int m2 = stoi(next.substr(3, 2));
        return (h2 - h1) * 60 + m2 - m1;
    }
};